### PR TITLE
Add memory hook for cheat/rumble/cheevos support

### PR DESCRIPTION
**Pull request by negativeExponent**

### CODE REVIEW

File: libblastem.c [hunk 1]
```diff
@@ -238,11 +242,43 @@ RETRO_API unsigned retro_get_region(void)
 /* Gets region of memory. */
 RETRO_API void *retro_get_memory_data(unsigned id)
 {
+	if (id == RETRO_MEMORY_SYSTEM_RAM)
```
Comment (Jamiras) [inline libblastem.c:245]: I'm not sure if it figures in to the cheat/rumble support, but cheevos also use `RETRO_MEMORY_SAVE_RAM`. Would be nice to add that while you're in here.

**Comment by hizzlekizzle** [comment]
I passed this PR to MoD upstream. I'll leave this open until I'm sure it's been committed and then I'll close it and re-sync the repo to pull it in.

**Comment by negativeExponent** [comment]
save memory hook added, at the same time implements save support and cheevos. not quite sure for some save types requiring byteswap (RAM_FLAG_BOTH flag) since there is no way to do this on realtime when using the save api. Works with Phantasy Star IV though.

File: libblastem.c (hunk 1 above)
Comment (negativeExponent) [reply libblastem.c]: thanks for the review. save memory added.

**Comment by negativeExponent** [comment]
this PR has since been merged in upstream and then synced into the libretro branch where we were suppose to work on. So aside from this was based off master branch, this it rather unneeded now and can be closed if its not needed anymore.

**Comment by hizzlekizzle** [comment]
alright, mission accomplished :)

**Pull request closed without merging**